### PR TITLE
Fix 2D array uint texture fragment shader

### PR DESCRIPTION
**Pull request by qkmiao**

### CODE REVIEW

File: sdk/tests/js/tests/tex-image-and-sub-image-utils.js [hunk 1]
```diff
@@ -163,7 +163,7 @@ var TexImageUtils = (function() {
     'in vec2 texCoord;',
     'out vec4 fragData;',
     'void main() {',
-    '    vec4 data = vec4(texture(tex, texCoord, 0).rgb 1.0);',
+    '    vec4 data = vec4(texture(tex, vec3(texCoord, 0.0)).rgb, 1.0);',
```
Comment (zhenyao) [inline sdk/tests/js/tests/tex-image-and-sub-image-utils.js]: This is incorrect.  For Sampler3dArray, it is texture(tex, xy, z(integer)), not texture(tex, xyz)

Comment (qkmiao) [inline sdk/tests/js/tests/tex-image-and-sub-image-utils.js]: GLSL spec 3.00.3 defines "gvec4 texture (gsampler2DArray sampler, vec3 P [, float bias] )". https://www.khronos.org/registry/gles/specs/3.0/GLSL_ES_Specification_3.00.3.pdf

**Comment by zhenyao** [comment]
Ah, you are right.  I don't know where I got the wrong idea. :(

File: sdk/tests/js/tests/tex-image-and-sub-image-utils.js (hunk 1 above)
Comment (kenrussell) [inline sdk/tests/js/tests/tex-image-and-sub-image-utils.js]: Would it not make more sense to define:

  uvec4 data = texture(tex, vec3(texCoord, 0.0));
  fragData = vec4(float(data[0])/255.0, float(data[1])/255.0, float(data[2])/255.0, 1.0);

The cases to float below aren't currently necessary given that data is a float vector.

**Comment by kenrussell** [comment]
LGTM overall after the fact, with one comment.

**Pull request merged**